scheduler/generic_sched.go

```diff
@@ -177,6 +179,7 @@ func (s *GenericScheduler) createBlockedEval(planFailure bool) error {
 // process is wrapped in retryMax to iteratively run the handler until we have no
 // further work or we've made the maximum number of attempts.
 func (s *GenericScheduler) process() (bool, error) {
+	s.queuedAllocs = make(map[string]int)
```

Comment:
Move bellow where we get the job, and if the job is not nil, initialize the size of the map to be equal to the number of task groups